staging/src/k8s.io/client-go/tools/cache/shared_informer.go

```diff
@@ -575,27 +590,35 @@ func (s *sharedIndexInformer) AddEventHandlerWithResyncPeriod(handler ResourceEv
 
 	handle := s.processor.addListener(listener)
 	for _, item := range s.indexer.List() {
-		listener.add(addNotification{newObj: item})
+		// Note that we deliver these notifications with the lock held
+		// and before returning the handle. That means there is never a
```

Comment:
> Note that we deliver these notifications with the lock held and before returning the handle

nit: I misunderstood this to mean the notifications were delivered *to the handler* before returning, and went off on a rabbit trail because I was pretty sure that wasn't true (spoiler: it's not, there's two async levels before the notifications get to the handler in pop() and run())

maybe rephrase to something like "we enqueue these notifications to the syncTracker with the lock held and before returning the handle" to save future us from rabbit trails

Reply:
Yeah sorry, "enqueue" is much less confusing, I'll change it

Reply:
fixed, sorry